greenpeace/planet4-plugin-gutenberg-blocks
# PLANET-4535: Split Timeline block scripts

Ref: https://jira.greenpeace.org/browse/PLANET-4535
Following https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/508/

We now split blocks each in their own script, instead of merging everything  in `(frontend|editor)Index.js`.

This PR mostly renames files, and adds a render function. The render function could be generalized and made as a dependency of other scripts (as it's gonna happen for hydration I guess ?).
- `TimelineScript.js` instantiate a Timeline on the frontend side
- `TimelineStyle.scss` is used for frontend instance
- `TimelineEditorScript.js` instantiate a Timeline editor, in the gutenberg editor
- `TimelineEditorStyle.scss` is used for editor specific styles

## Test

Timeline block should work as usual.
Here's a googlesheet to test https://docs.google.com/spreadsheets/d/1XruuRzjiu4t2ia7NaBXMqItr4NAbc7ctcolg1uhlIL4/edit#gid=0
- Create a page, insert a Timeline block with the test spreadsheet
- Check in your browser network console that Timeline scripts are included individually:
  ![Screenshot from 2021-04-19 09-13-14](https://user-images.githubusercontent.com/617346/115196070-cb196880-a0ef-11eb-99f8-d36c257eefa8.png)
- On preview/published page, check that Timeline scripts are included individually
![Screenshot from 2021-04-19 09-16-55](https://user-images.githubusercontent.com/617346/115196339-16cc1200-a0f0-11eb-9409-2a6109aa76fb.png)
- Timeline scripts shouldn't be included in any page without a Timeline in it (default Homepage for example)
